cas-mfa-java/src/main/java/net/unicon/cas/mfa/authentication/DefaultRegisteredServiceMfaRoleProcessorImpl.java

```diff
@@ -170,7 +174,7 @@ private MultiFactorAuthenticationRequestContext getMfaRequestContext(final Servi
                     serviceMfaData.getAuthenticationMethod()).getRank();
             final MultiFactorAuthenticationSupportingWebApplicationService svc =
                     this.mfaServiceFactory.create(targetService.getId(), targetService.getId(),
-                            targetService.getArtifactId(), serviceMfaData.getAuthenticationMethod(),
+                            targetService.getArtifactId(), "POST".equals(method) ? ResponseType.POST : ResponseType.REDIRECT, serviceMfaData.getAuthenticationMethod(),
```

Comment:
I'm completely guessing here in this method because I'm not clear on the role of this class or the role of the `PrincipalAttributeMultiFactorAuthenticationRequestResolver` either. If the role of the `ArgumentExtractor` is to create the service, why then do these two classes need to create the service as well?